elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLColumn.java

```diff
@@ -5,18 +5,131 @@
+import com.yahoo.elide.datastores.aggregation.metadata.models.Column;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.SQLQueryEngine;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.annotation.DimensionFormula;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.annotation.JoinTo;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
 /**
- * Column with physical SQL information like reference and join to path.
+ * SQLColumn is a wrapper for {@link Column} that contains physical information for {@link SQLQueryEngine}.
  */
 public interface SQLColumn {
-    Table getTable();
-
-    String getName();
+    /**
+     * Get wrapped column
+     *
+     * @return column
+     */
+    Column getColumn();
 
+    /**
+     * Get full sql reference
+     *
+     * @return physical reference
+     */
     String getReference();
 
-    JoinPath getJoinPath();
+    /**
+     * Get join paths needed for this column
+     *
+     * @return all join paths to this column
+     */
+    List<JoinPath> getJoinPaths();
+
+    /**
+     * SQL Column would resolve references in {@link DimensionFormula} when constructing physical reference.
+     *
+     * @return {@link LabelResolver} for sql columns
+     */
+    default LabelResolver constructSQLColumnLabelResolver(EntityDictionary dictionary) {
+        Class<?> tableClass = dictionary.getEntityClass(getColumn().getTable().getId());
+        String fieldName = getColumn().getName();
+
+        DimensionFormula formula = dictionary.getAttributeOrRelationAnnotation(
+                tableClass, DimensionFormula.class, fieldName);
+
+        if (formula == null) {
+            JoinTo joinTo = dictionary.getAttributeOrRelationAnnotation(tableClass, JoinTo.class, fieldName);
+
+            return joinTo == null || joinTo.path().equals("")
+                    ? getLogicalColumnResolver(tableClass, fieldName)
+                    : getJoinToResolver(tableClass, joinTo);
+        } else {
+            return getFormulaResolver(tableClass, formula);
+        }
+    }
+
+    /**
+     * Get a logical column reference resolver
+     *
+     * @param tableClass table class
+     * @param fieldName field name
+     * @return a resolver
+     */
+    default LabelResolver getLogicalColumnResolver(Class<?> tableClass, String fieldName) {
+        return new LabelResolver(getColumn()) {
+            @Override
+            public String resolveLabel(LabelStore labelStore, String tableAlias) {
+                return getFieldAlias(
+                        tableAlias,
+                        labelStore.getDictionary().getAnnotatedColumnName(tableClass, fieldName));
+            }
+        };
+    }
+
+    /**
+     * Get a {@link JoinTo} reference resolver.
+     *
+     * @param tableClass table class
+     * @param joinTo join to path
+     * @return a resolver
+     */
+    default LabelResolver getJoinToResolver(Class<?> tableClass, JoinTo joinTo) {
```

Comment:
I assume we can collapse this if we move Column level logic from column to SQLColumn